src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -266,6 +281,36 @@ impl TargetInfo {
                 res
             };
 
+            let target_spec_json = {
+                let mut res: String = String::new();
+                loop {
+                    match lines.next() {
+                        Some(line) if line == "}" => {
+                            res.push_str("}");
+                            break;
+                        }
+                        Some(line) => res.push_str(line.into()),
+                        None => {
+                            return error_missing_print_output(
+                                "target-spec-json",
+                                &process,
+                                &output,
+                                &error,
+                            )
+                        }
+                    }
+                }
+                res
+            };
+
+            let target_spec: TargetSpec = serde_json::from_str(&target_spec_json.as_str())
+                .with_context(|| {
+                    format!(
+                    "failed to parse target spec from `rustc --print=target-spec-json`, got:\n{}",
+                    output
+                )
+                })?;
```

Comment:
can you please make this non-fatal, such that you just keep a `None` if the format doesn't match what you expect (and probably also when not using the nightly channel, the `RUSTC_BOOTSTRAP=1` is quite problematic above and should be avoided), and then avoid emitting any error in that case.